### PR TITLE
Clean/api.001

### PR DESCRIPTION
**Pull request by esguerra**
Move all api functions under just one app hierarchy.

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -737,8 +736,8 @@ class StructureAssignGenericNumbers(views.APIView):
 
     def post(self, request):
 
-        # root, ext = os.path.splitext(request._request.FILES['pdb_file'].name)
-        generic_numbering = GenericNumbering(StringIO(request._request.FILES['pdb_file'].file.read().decode('UTF-8',"ignore")))
+        root, ext = os.path.splitext(request.FILES['pdb_file'].name)
```
Comment (AJK-dev) [inline api/views.py:739]: Not necessary?

File: api/views.py [hunk 2]
```diff
@@ -737,8 +736,8 @@ class StructureAssignGenericNumbers(views.APIView):
...
+        generic_numbering = GenericNumbering(StringIO(request.FILES['pdb_file'].file.read().decode('UTF-8',"ignore")))
```
Comment (AJK-dev) [inline api/views.py:740]: Why is request._request not necessary anymore?
I added this recently to fix the functioning of these calls on the server with PR #172 .

**Review by AJK-dev — COMMENTED** [review]
Two open questions

File: api/views.py [hunk 3]
```diff
@@ -757,13 +756,13 @@ class StructureSequenceParser(views.APIView):
     curl -X POST -F "pdb_file=@myfile.pdb" http://gpcrdb.org/services/structure/parse_pdb
     """
     parser_classes = (FileUploadParser,)
-    renderer_classes = (JSONRenderer, )
+    renderer_classes =(JSONRenderer)
```
Comment (AJK-dev) [inline api/views.py:759]: Small note by example:
> a = (1,)
> type(a)
OUTPUT: <class 'tuple'>

> a = (1)
> type(a)
OUTPUT: <class 'int'>

File: api/urls.py [hunk 4]
```diff
@@ -68,5 +68,7 @@
     url(r'^species/$', views.SpeciesList.as_view(), name='species-list'),
     url(r'^species/(?P<latin_name>[^/]+)/$', views.SpeciesDetail.as_view(), name='species-detail'),
     url(r'^mutants/(?P<entry_name>[^/].+)/$', views.MutantList.as_view(), name='mutants'),
-    url(r'^drugs/(?P<entry_name>[^/].+)/$', views.DrugList.as_view(), name='drugs')
+    url(r'^drugs/(?P<entry_name>[^/].+)/$', views.DrugList.as_view(), name='drugs'),
+    url(r'^plot/helixbox/(?P<entry_name>[^/].+)/$', views.HelixBoxView.as_view(), name='helixbox'),
```
Comment (AJK-dev) [inline api/urls.py:72]: With regard to the URLs: when cleaning the API calls this should (afaik) also be changed in protwis/urls.py and then api_gpcr can be removed right?

Comment (esguerra) [reply api/urls.py:72]: You are right. I will look into it and change it.

**Review by AJK-dev — CHANGES_REQUESTED** [review]
All in order, if you also remove the api_gpcr folder then this PR is complete.

**Comment by esguerra** [comment]
> All in order, if you also remove the api_gpcr folder then this PR is complete.

api_gpcr folder removed.
Whenever we move to djangorestframework >= 3.10 
[https://www.django-rest-framework.org/community/3.10-announcement/](https://www.django-rest-framework.org/community/3.10-announcement/)
At the moment we are in djangorestframework==3.9.3

**Pull request merged**